File: framework/tests/test-application/app/controllers/Application.java
package controllers;

import play.mvc.Controller;
import play.Logger;
import play.Invoker;
import play.mvc.Before;
import java.util.HashSet;
import java.util.Date;

public class Application extends Controller {

    @Before
    static void log(String name, Integer age) {
        Logger.info(">>>> %s", name);
        if ("Guillaume".equals(name)) {
            redirect("http://www.amazon.com");
        }
        models.Test.test();
        if(name != null) {
            session.put("name", name);
        }  
    }

    public static void index(String name, Integer age, HashSet<Date> date) { 
        models.Test t = new models.Test();
        t.prop = "kiki";
        Yop.kiki();
        renderText("%s | SHIGETA フローラルウォーターは (%s -> %s / %s)", t.prop, name, age, date);        
    }

    public static void test() {
        Orders.test("Guillaume");
    }
    
}

File: samples-and-tests/with-scala/app/controllers/Application.scala
package controllers 

import play._
import play.mvc._
import play.db.jpa._
import play.data.validation._
import play.libs._

import models._

object Application extends Actions {
    
    @Before
    private def check {
        println("Check ... " + configuration("yop", "nf") )
        renderArgs += ("kiki" -> 9)
    }
    
    def test {        
        response <<< OK
        response <<< "text/plain"
        response <<< ("X-Test" -> "Yop")
        response <<< """|Hello World
                        |
                        |My Name is Guillaume""".stripMargin        
    }
    
    def test2 {        
        response <<< NOT_FOUND
        response <<< "text/html"
        response <<< <h1>Not found, sorry</h1>        
    }
    
    def index(@Min(10) nimp: Int = 5, @Required name: String = "Guillaume") {
        println(nimp)
        println(request.path)
        val age = 59 
        var yop = 8
        yop = yop + 3
        println(name)
        
        info("Yop %d", 9)
        
        val users = find[User]("byPassword", "88style").fetch
        
        response <<< OK
        response <<< "YOUHOU" 
        response <<< "X-Yop" -> "hope"
        
        render(name, age, yop, users)
    }
    
    def addOne() {
        val user = new User("guillaume@gmail.com", "88style", "Guillaume")
        user.save()
        index()
    }
    
    def goJojo() {
        index(name="Jojo") 
    }
    
    def api = renderXml(<items><item id="3">Yop</item></items>) 
    
    def yop = render("@index") 
    
    def helloWorld = <h1>Hello world!</h1>
    
    def hello(name: String) = <h1>Hello { if(name != null) name else "Guest" }!</h1>
    
    def nbOfUsers = User.size
    
    def captcha = Images.captcha
    
    @After
    private def log {
        println(new User("tom@gmail.com", "secret", "Tom"))
        println(new Company)
        //renderText("Oops")
    }
    
}
